mllib/src/main/scala/org/apache/spark/mllib/tree/GradientBoostedTrees.scala

```diff
@@ -250,7 +250,8 @@ object GradientBoostedTrees extends Logging {
       predError = GradientBoostedTreesModel.updatePredictionError(
         input, predError, baseLearnerWeights(m), baseLearners(m), loss)
       predErrorCheckpointer.update(predError)
-      logDebug("error of gbt = " + predError.values.mean())
+      val currentPredError = predError.values.mean()
```

Comment:
Since this is only used in logDebug, it would be better to put it inside logDebug so that it is not computed if debug logging is turned off.